ecosystem/node-checker/src/main.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+// For use while we're developing.
+#![allow(dead_code)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+
+mod args;
+mod build_evaluators;
+mod metric_collector;
+mod metric_evaluator;
+mod public_types;
+mod runner;
+
+use anyhow::{anyhow, bail, Context, Result};
+use args::Args;
+use build_evaluators::build_evaluators;
+use clap::Parser;
+use log::{debug, info};
+use metric_collector::{MetricCollector, ReqwestMetricCollector};
+use metric_evaluator::MetricsEvaluator;
+use poem::{
+    handler, http::StatusCode, listener::TcpListener, Error as PoemError, Result as PoemResult,
+    Route, Server,
+};
+use poem_openapi::{payload::Json, OpenApi, OpenApiService};
+use public_types::{CompleteEvaluation, NodeUrl};
+use reqwest::Client as ReqwestClient;
+use runner::{BlockingRunner, BlockingRunnerArgs, Runner};
+use std::{collections::HashSet, hash::Hash, path::PathBuf, sync::Arc};
+use url::Url;
+
+// TODO: Replace this with the real frontend, or perhaps an error handler if we
+// decide to route the frontend to just a static hoster such as nginx.
+#[handler]
+fn root() -> String {
+    "Hello World!".to_string()
+}
+
+struct Api<M: MetricCollector, R: Runner> {
+    pub runner: R,
+    pub target_metric_collector: Option<M>,
+    pub allow_preconfigured_test_node_only: bool,
+}
+
+// I choose to keep both methods rather than making these two separate APIs because it'll
+// make for more descriptive error messages.
+#[OpenApi]
+impl<M: MetricCollector, R: Runner> Api<M, R> {
+    #[oai(path = "/check_node", method = "get")]
+    async fn check_node(&self, target_node: Json<NodeUrl>) -> PoemResult<Json<CompleteEvaluation>> {
+        if self.allow_preconfigured_test_node_only {
```

Comment:
Can we rename this? I find the term `test` confusing. Why not just `allow_preconfigured_node_only`?

Reply:
Test is a little confusing agreed, but I want to qualify it with something to make it clear that this is the node being tested. I've toyed with "investigation" and "target" but they don't seem much better.

Reply:
Yeah, the other options aren't great. I still think `allow_preconfigured_node_only` is clearer than `allow_preconfigured_test_node_only`, but I'll leave it up to you.